beacon-chain/rpc/eth/helpers/lightclient/lightclient.go

```diff
@@ -0,0 +1,240 @@
+package lightclient
+
+import (
+	"bytes"
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	types "github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+	ethpbv1 "github.com/prysmaticlabs/prysm/v4/proto/eth/v1"
+	ethpbv2 "github.com/prysmaticlabs/prysm/v4/proto/eth/v2"
+	"github.com/prysmaticlabs/prysm/v4/proto/migration"
+)
+
+func NewLightClientOptimisticUpdateFromBeaconState(
+	ctx context.Context,
+	config *params.BeaconChainConfig,
+	state state.BeaconState,
+	block interfaces.ReadOnlySignedBeaconBlock,
+	attestedState state.BeaconState) (*ethpbv2.LightClientUpdate, error) {
+	// assert compute_epoch_at_slot(attested_state.slot) >= ALTAIR_FORK_EPOCH
+	attestedEpoch := types.Epoch(uint64(attestedState.Slot()) / uint64(config.SlotsPerEpoch))
+	if attestedEpoch < types.Epoch(config.AltairForkEpoch) {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid attested epoch: %d", attestedEpoch)
```

Comment:
It would be better to return `RpcError` (defined in `beacon-chain/rpc/core/errors.go`) everywhere instead of `status` errors, which are a gRPC construct.

Reply:
I did remove the grpc status and codes from the handlers but these ones I missed. Will replace them.

Reply:
After replacing the grpc types with `core.RpcError` I got myself into a circular dependency:
- the LC helper depends on rpc.core
- rpc.core depends on blockchain
- blockchain depends on LC helper (this is added later in the series)

Easiest thing I can think of is moving the rpc/core/errors.go to a "submodule" like `rpc.core.errors`. That will only affect my lc helper and `beacon-chain/rpc/core/validator.go`. 

What do you think?

Reply:
RPC is the outer layer of the system. It should not depend on the blockchain package. If your design requires this, then these helper functions should also be under blockchain. I think a new `blockchain/lightclient` package would make most sense. I haven't reviewed the logic of these functions yet, but they don't seem to be closely connected with any rpc functionality. Such a change would also mean that regular go errors will be returned from these functions.

Reply:
The LC helpers introduced in this PR do not depend on the blockchain package. The blockchain package will depend on the LC helpers in a future PR. 

However, `rpc.core` does depend on the `blockchain` package and that's where the circular dependency comes from. 

Not adding a dependency from the LC helpers on `rpc.core` will solve the issue. I think using regular errors will do the trick regardless of where the LC helpers live.

Reply:
I want to avoid the blockchain package depending on anything in the rpc package, if possible.